core/src/main/java/org/apache/iceberg/hadoop/HadoopMetricsContext.java

```diff
@@ -114,12 +111,15 @@ public void increment(Integer amount) {
     };
   }
 
-  private void writeObject(ObjectOutputStream out) throws IOException {
-    out.defaultWriteObject();
-  }
+  private FileSystem.Statistics statistics() {
+    if (statistics == null) {
```

Comment:
I think this field needs to be volatile by double-check locking pattern, because otherwise it may return partially initialized statistics object in the not null case?

Reply:
Thanks, missed that corrected it.